api/webhook/post.go

```diff
@@ -207,6 +207,23 @@ func PostWebhook(c *gin.Context) {
 		return
 	}
 
+	// attach a sender SCM id if the webhook payload from the SCM has no sender id
+	if len(b.GetSenderSCMID()) == 0 || b.GetSenderSCMID() == "0" {
+		// fetch scm user id for pusher
+		senderID, err := scm.FromContext(c).GetUserID(ctx, b.GetSender(), repo.GetOwner().GetToken())
+		if err != nil {
+			retErr := fmt.Errorf("unable to assign sender SCM id: %w", err)
+			util.HandleError(c, http.StatusBadRequest, retErr)
+
+			h.SetStatus(constants.StatusFailure)
+			h.SetError(retErr.Error())
+
+			return
+		}
+
+		b.SetSenderSCMID(senderID)
+	}
```

Comment:
in what circumstances will this get hit? or, when is there not a SCMID? Just trying to gauge overhead.

Reply:
this might happen when a git webhook payload doesnt contain a `sender` object and we fallback to setting sender using a field that contains username but no ID, for example `pusher` in this piece of code has no `id`

here: https://github.com/go-vela/server/blob/main/scm/github/webhook.go#L179

though, when can _that_ scenario occur? i dont know, but the code is there, so it implies that its possible for sender to be empty. in this case the sender_scm_id from `GetID()` above will return `0`, hence the `== '0'`.

ill add some comments real quick

Reply:
yup, sounds good. it didn't seem like it would get hit a lot. thanks for the clarification.